platform/engineering/request-for-comment/2020-11-4-storybook-hosting.md

```diff
@@ -0,0 +1,70 @@
+# RFC: Host Storybook on design.va.gov
+
+- Date: 2020-11-4
+- Related Issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/15068
+
+## Background
+We have component library documentation in two places:
+- https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design
+- https://design.va.gov/components
+
+We're going to use Storybook to consolidate the documentation into one place:
+design.va.gov.
+
+design.va.gov is a Jekyll site. The documentation in it is manually updated. The
+repo can be found at
+[`vets-design-system-documentation`](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/).
+It's deployed daily by <this Jenkins job> to the S3 bucket from which the static
+site is served.
+
+## Motivation
+We want to serve the component library documentation from Storybook on
+design.va.gov to ensure the documentation is in a single place that's
+automatically updated when updates are made to the library. We chose Storybook
+as the tool to keep the documentation closely tied to the code itself.
+
+## Scope
+Outside the scope of this document is the code location of the component library
+and Storybook. While the location of the code will impact where the CI
+configuration is kept, the solution outlined here must be possible whether the
+code lives in
+[`veteran-facing-services-tools`](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools),
+[`vets-website`](https://github.com/department-of-veterans-affairs/vets-website),
+or its own repo, as we may move it later.
+
+For now, however, we're planning on moving the component library to its own repo.
+
+## Design
+- Storybook will be deployed to the S3 bucket from which design.va.gov is served
+  - **Question:** How will we be able to ensure deploying the Jekyll site won't
```

Comment:
Probably have to build the whole thing like we do on vets-website or download / cache what's on S3 before syncing.

Reply:
+1 to @rianfowler's comment above. I'm sure you already have this link, but for my own future reference: https://storybook.js.org/docs/react/workflows/publish-storybook

Reply:
Right, so once we build the whole thing with `yarn build-storybook`, it'll create all the files needed to serve the Storybook. We can push those files up to the design.va.gov S3 bucket at `components/`, but if we do that in a separate job from the Jekyll site deploy, when that Jekyll site deploy job runs, will it nuke the contents of that directory? How do we get it to ignore that directory?

Looks like there's an `--exclude` option to the `aws s3 sync` command, but I don't know how to use it. Would that do the trick?

Reply:
If all else fails we could build storybook for each full Jekyll deploy.